optuna/integration/sklearn.py

```diff
@@ -4,47 +4,38 @@
 from numbers import Integral
 from numbers import Number
 from time import time
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import Union
 
 import numpy as np
 import scipy as sp
+from scipy.sparse import spmatrix
 
 from optuna._experimental import experimental
 from optuna._imports import try_import
-from optuna import distributions  # NOQA
-from optuna import TrialPruned  # NOQA
-from optuna import logging  # NOQA
-from optuna import samplers  # NOQA
-from optuna import study as study_module  # NOQA
-from optuna.study import StudyDirection  # NOQA
-from optuna import trial as trial_module  # NOQA
-from optuna.trial import FrozenTrial  # NOQA
-from optuna import type_checking  # NOQA
-
-if type_checking.TYPE_CHECKING:
-    import pandas as pd  # NOQA
-    from scipy.sparse import spmatrix  # NOQA
-    from typing import Any  # NOQA
-    from typing import Callable  # NOQA
-    from typing import Dict  # NOQA
-    from typing import Iterable  # NOQA
-    from typing import List  # NOQA
-    from typing import Mapping  # NOQA
-    from typing import Optional  # NOQA
-    from typing import Union  # NOQA
-
-    ArrayLikeType = Union[List, np.ndarray, pd.Series, spmatrix]
-    OneDimArrayLikeType = Union[List[float], np.ndarray, pd.Series]
-    TwoDimArrayLikeType = Union[List[List[float]], np.ndarray, pd.DataFrame, spmatrix]
-    IterableType = Union[List, pd.DataFrame, np.ndarray, pd.Series, spmatrix, None]
-    IndexableType = Union[Iterable, None]
+from optuna import distributions
+from optuna import logging
+from optuna import samplers
+from optuna import study as study_module
+from optuna.study import StudyDirection
+from optuna.trial import FrozenTrial
+from optuna.trial import Trial
+from optuna import TrialPruned
 
 with try_import() as _imports:
+    import pandas as pd
```

Comment:
When ``pandas`` is not installed, python exits this block before importing ``sklearn`` even if ``sklearn`` is installed. I recommend splitting the try_import block as follows.

```python
with try_import:
    import pandas as pd

with try_import() as _imports:
    ...
```

Reply:
I see. Thank you again for detailed comments.